### ISSUE TITLE
s3: Setup s3 server access logging target bucket policy

### ISSUE THREAD

**Issue by cmckni3** [aws/aws-cdk#6545]
Automatically setup bucket policy when enabling S3 server access logs via `Bucket` construct when setting up [access logging](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.Bucket.html#serveraccesslogsbucket)

https://docs.aws.amazon.com/AmazonS3/latest/dev/ServerLogs.html#server-access-logging-overview

### Use Case

- S3 server access log delivery

### Proposed Solution

N/A

### Other

---

This is a :rocket: Feature Request

**Comment by cmckni3** [comment]
Maybe constructs should generally support adding the log delivery service.

Looks like the s3 construct does not add it based on [Add S3 server access logs PR](https://github.com/aws/aws-cdk/pull/5072).

Feedback welcome. Let me know if you want me to update the requirements/spec of this.

**Comment by rix0rrr** [comment]
> It's actually a generic AWS logs delivery account for the service `delivery.logs.amazonaws.com`.

Wait, if that's true then why don't we use:

```
"Service": "delivery.log.amazonaws.com"
```

Everywhere instead of the account numbers?

**Comment by rix0rrr** [comment]
> Let me know if you want me to update the requirements/spec of this.

A little please. Your ticket is asking for a specific solution you have in mind, it's not really describing what you're trying to achieve and the problem you're running into.

We generally prefer starting from the use case/problem because it gives us more flexibility searching the solution space, at the same time evaluating whether the fixes we have in mind might be solving your use case while solving multiple use cases at once, or doing so more elegantly, or whatnot.

For example, it seems to me your use case is:

> I want to configure S3 access logging

Your problem is:

> I don't have access to the account numbers I need to correctly configure access to the bucket

Your suggested immediate solution is:

> Give me the account numbers so that I can configure permissions on the bucket

But another (arguably more CDK-like solution) would be:

> The permissions on the logging bucket are configured automatically.

Let me know if I've got that right, and let me know if there's a particular reason you rejected the "it just works" solution out of hand? Might be useful to let us know about constraints you have or problems you might see.

--------

Implementor's note: these accounts should probably go into `@aws-cdk/region-info`.

**Comment by cmckni3** [comment]
Got it and that's what I figured.

I updated the issue to add the policies for `delivery.logs.amazonaws.com` (`s3:GetBucketAcl` and `s3:PutObject`) when enabling s3 server access logging.

I was very confused by the various ways that logging can be enabled for different services. Some use ACLs (legacy) but most only require bucket policies.

Correct me if I'm wrong but I believe ELB is the only special case for bucket policies requiring the [account id](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#access-logging-bucket-permissions).
